artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java

```diff
@@ -772,6 +772,10 @@ public void recreateConsumerOnServer(ClientConsumerInternal consumerInternal,
       if (!queueInfo.isDurable()) {
          CreateQueueMessage_V2 createQueueRequest = new CreateQueueMessage_V2(queueInfo.getAddress(), queueInfo.getName(), queueInfo.getRoutingType(), queueInfo.getFilterString(), false, queueInfo.isTemporary(), queueInfo.getMaxConsumers(), queueInfo.isPurgeOnNoConsumers(), queueInfo.isAutoCreated(), false, queueInfo.isExclusive(), queueInfo.isLastValue());
 
+         sendPacketWithoutLock(sessionChannel, createQueueRequest);
+      } else if (queueInfo.isAutoCreated()) {
```

Comment:
seems almost duplicated code lines.

why not make the if statement of the if above (!queueInfo.isDurable() || queueInfo.isAutoCreated()) 
and replace the only difference which is the hardcoded false for durability to take queueInfo.isDurable().

Reply:
Will do, @michaelandrepearce. That's a much more elegant solution.